source/scripts/data/monsters.js

```diff
@@ -31,4 +31,206 @@ export default {
+                    delete choices[choices.indexOf({x: +1})]
+            }
+            for(var monster of this.game.monsters) {
+                if (monster.position.y == this.position.y - 1 && monster.position.x == this.position.x)
+                    delete choices[choices.indexOf({y: -1})]
+            }
+            for(var monster of this.game.monsters) {
+                if (monster.position.y == this.position.y + 1 && monster.position.x == this.position.x)
+                    delete choices[choices.indexOf({y: +1})]
+            }
+
+            if (this.position.x == 0)
+                delete choices[choices.indexOf({x: -1})]
+            if (this.position.x == DATA.FRAME.WIDTH)
+                delete choices[choices.indexOf({x: +1})]
+            if (this.position.y == 0)
+                delete choices[choices.indexOf({y: -1})]
+            if (this.position.y == DATA.FRAME.HEIGHT)
+                delete choices[choices.indexOf({y: +1})]
+
+            return choices[Math.floor((Math.random() * choices.length))]
+        }
+    },
+    BLUE_BAT: {
+        sprite: DATA.SPRITES.MONSTERS.BAT,
+        color: DATA.COLORS.BLUE,
+        health: 1,
+        strength: 1,
+        movement: function () {
+            if (this.position.x < 0) return {x: +1}
+            if (this.position.x >= DATA.FRAME.WIDTH) return {x: -1}
+            if (this.position.y < 0) return {y: +1}
+            if (this.position.y >= DATA.FRAME.HEIGHT) return {y: -1}
+
+            var choices = [
+                {x: -1},
+                {x: +1},
+                {y: -1},
+                {y: +1}
+            ]
+
+            for(var monster of this.game.monsters) {
+                if (monster.position.x == this.position.x - 1 && monster.position.y == this.position.y)
+                    delete choices[choices.indexOf({x: -1})]
+            }
+            for(var monster of this.game.monsters) {
+                if (monster.position.x == this.position.x + 1 && monster.position.y == this.position.y)
+                    delete choices[choices.indexOf({x: +1})]
+            }
+            for(var monster of this.game.monsters) {
+                if (monster.position.y == this.position.y - 1 && monster.position.x == this.position.x)
+                    delete choices[choices.indexOf({y: -1})]
+            }
+            for(var monster of this.game.monsters) {
+                if (monster.position.y == this.position.y + 1 && monster.position.x == this.position.x)
+                    delete choices[choices.indexOf({y: +1})]
+            }
+
+            if (this.position.x == 0)
+                delete choices[choices.indexOf({x: -1})]
+            if (this.position.x == DATA.FRAME.WIDTH)
+                delete choices[choices.indexOf({x: +1})]
+            if (this.position.y == 0)
+                delete choices[choices.indexOf({y: -1})]
+            if (this.position.y == DATA.FRAME.HEIGHT)
+                delete choices[choices.indexOf({y: +1})]
```

Comment:
Sounds like a plan. I think I've got this nailed down (until we get the "random access" happening, at least).